### DISCUSSION TITLE
different cell color for the same file, bug or something else?

### DISCUSSION THREAD

**Discussion by DanieleAsrael** [PHPOffice/PhpSpreadsheet#4194] [General]
Hi all,

on an old installation I have phpoffice/phpexcel 1.8.2.
Now we have upgraded everything including php version and I am using
phpoffice/phpspreadsheet 1.29.2, on a new server.

This piece of code gives me a different color for the same file, how is it possible?
Does anyone have an explanation?

```
$cell = $sheet->getCellByColumnAndRow($col, $row);
$fill = $cell->getStyle()->getFill()->getStartColor()->getARGB();
```

As result I have:
```
$fill = "FFE2EEDA"; // (phpexcel 1.8.2)
$fill = "FFE2F0D9"; // (phpspreadsheet 1.29.2)
```

Thanks in advance for help!

**Comment by oleibman** [comment]
No idea without seeing the file (or the code you used to set the color). Regardless of the cause, this seems to be an extremely small difference - is it causing you a problem?

**Reply by DanieleAsrael** [reply]
Ok, I wrote two script with those version

**phpexcel 1.8.2 - PHP 7.4.33**
composer require phpoffice/phpexcel
```
<?php

require_once 'vendor/autoload.php';
$filePath   = "bbb.xlsx";
$xls_reader = PHPExcel_IOFactory::createReader('Excel2007');

$xls                = $xls_reader->load($filePath);
$sheet              = $xls->getActiveSheet();
$highestRow         = $sheet->getHighestRow();
$highestColumn      = $sheet->getHighestColumn();
$highestColumnIndex = PHPExcel_Cell::columnIndexFromString($highestColumn);

foreach (range('C', $highestColumn) as $col)
{
    $offer = [];
    for ($row = 1; $row <= $highestRow; $row++)
    {
        /** @var PHPExcel_Cell */
        $cell = $sheet->getCell($col.$row);

        $fill  = $cell->getStyle()->getFill()->getStartColor()->getARGB();
        $value = $cell->getValue();
        echo "$col|$row|$fill|$value\n";
    }
}
```

**phpspreadsheet 1.29.2 - PHP 8.2.24**
composer require phpoffice/phpspreadsheet
```
<?php

use PhpOffice\PhpSpreadsheet\IOFactory;

require_once 'vendor/autoload.php';
$filePath = "bbb.xlsx";
$handle   = IOFactory::load($filePath);

$sheet         = $handle->getActiveSheet();
$highestRow    = $sheet->getHighestRow();
$highestColumn = $sheet->getHighestColumn();

foreach (range('C', $highestColumn) as $col)
{
    $offer = [];
    for ($row = 1; $row <= $highestRow; $row++)
    {
        $cell = $sheet->getCell($col.$row);

        $fill  = $cell->getStyle()->getFill()->getStartColor()->getARGB();
        $value = $cell->getValue();
        echo "$col|$row|$fill|$value\n";
    }
}
```
output of these script are in attach, with the xlsx used.


P.S Yes, the difference caused a bug (nothing serious), who wrote the first version used background color to detect updated cell.
[phpexcel.log](https://github.com/user-attachments/files/17459149/phpexcel.log)
[phpspreadsheet.log](https://github.com/user-attachments/files/17459150/phpspreadsheet.log)
[bbb.xlsx](https://github.com/user-attachments/files/17459152/bbb.xlsx)

**Comment by oleibman** [comment]
Thank you for the sample files and the excellent documentation. The cells in question use a base color to which a "tint" is applied. PR #3580 changed the calculation used for handling the tint to be a bit more accurate than the calculation that had been used in PHPExcel. That is why you are seeing this difference. As you can see from the text for this change, very small differences like this are to be expected, but large differences between PhpSpreadsheet and Excel, like the issue that prompted that PR, go away. Sorry that it caused you a problem.

**Reply by DanieleAsrael** [reply]
Thank you so much for your excellent work on the library and for the clarification regarding the issue. I appreciate your thorough explanation about the changes in the tint calculation.

I would also like to reassure you that the problem caused by this difference has not been very impactful on my work.

Thanks again!
